app/questionnaire/questionnaire_store_updater.py

```diff
@@ -321,6 +321,20 @@ def _capture_section_dependencies_for_answer(self, answer_id: str) -> None:
             else:
                 self.dependent_sections.add(DependentSection(section_id, None, None))
 
+    def _capture_section_dependencies_for_current_section(self) -> None:
+        """Captures a unique list of section ids that are dependents of the current section."""
+
+        section_id_section_dependents: set[
+            str
+        ] = self._schema.when_rules_section_dependencies_by_section.get(
+            self._current_location.section_id, set()
+        )
+
+        for dependent_section_id in section_id_section_dependents:
```

Comment:
I dont think this is no longer correc. Since you changed `_get_rules_section_dependencies`, the section ids here will include ids for calculated summary/ answer sources etc not just progress source. This only need to be done currently for progress source dependencies?.

Reply:
Separated the concerns by adding attribute `when_rules_section_dependencies_by_section_for_progress_value_source` ([here](https://github.com/ONSdigital/eq-questionnaire-runner/blob/66b2aca3b01704a57c64274944e274308414987c/app/questionnaire/questionnaire_schema.py#L90-L96))

If now iterates of this one specifically when capturing dependent sections for progress value sources ([here](https://github.com/ONSdigital/eq-questionnaire-runner/blob/66b2aca3b01704a57c64274944e274308414987c/app/questionnaire/questionnaire_store_updater.py#L337-L342))

When building the routing path, all the when dependencies are merged together ([here](https://github.com/ONSdigital/eq-questionnaire-runner/blob/66b2aca3b01704a57c64274944e274308414987c/app/questionnaire/path_finder.py#L67-L73))